File: vortexdbg-api/src/main/java/com/vortexdbg/signal/SignalTask.java
package com.vortexdbg.signal;

import com.vortexdbg.AbstractEmulator;
import com.vortexdbg.thread.RunnableTask;

public interface SignalTask extends RunnableTask {

    Number callHandler(SignalOps signalOps, AbstractEmulator<?> emulator);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/signal/SignalTask.kt
package com.vortexdbg.signal

import com.vortexdbg.AbstractEmulator
import com.vortexdbg.thread.RunnableTask

interface SignalTask : RunnableTask {

    fun callHandler(signalOps: SignalOps, emulator: AbstractEmulator<*>): Number

}
